filecoin/retriever.go

```diff
@@ -412,6 +412,13 @@ func (retriever *Retriever) queryCandidates(ctx context.Context, cid cid.Cid, ca
 		go func(i int, candidate RetrievalCandidate) {
 			defer wg.Done()
 
+			minerCfgs := retriever.config.MinerConfigs[candidate.MinerPeer.ID]
+			if minerCfgs.RetrievalTimeout != 0 {
+				var cancelFunc func()
+				ctx, cancelFunc = context.WithDeadline(ctx, time.Now().Add(minerCfgs.RetrievalTimeout))
```

Comment:
probably just context.WithTimeout is more relevant here no? (since what you have is a duration)

Reply:
well I went with deadline because filclient detects its presence and calls `socket.SetDeadline()` if there is one, so my assumption is that this is the right way to pass it on? https://github.com/application-research/filclient/blob/cb126018ee48269269ceef8c5d39d3bf0806f5b3/filclient.go#L466-L470

Reply:
nice